pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -395,25 +395,17 @@ func (b *Botanist) applyAndWaitForShootOperatingSystemConfig(ctx context.Context
 	}, nil
 }
 
-// generateCloudConfigExecutionChart renders the gardener-resource-manager configuration for the cloud config user data.
-// After that it creates a ManagedResource CRD that references the rendered manifests and creates it.
-func (b *Botanist) generateCloudConfigExecutionChart() (*chartrenderer.RenderedChart, error) {
-	bootstrapTokenSecret, err := kutil.ComputeBootstrapToken(context.TODO(), b.K8sShootClient.Client(), utils.ComputeSHA256Hex([]byte(time.Now().Format("2006-01-02")))[:6], "A bootstrap token generated by Gardener.", 48*time.Hour)
-	if err != nil {
-		return nil, err
-	}
-	workers := make([]map[string]interface{}, 0, len(b.Shoot.Info.Spec.Provider.Workers))
-	for _, worker := range b.Shoot.Info.Spec.Provider.Workers {
-
+// getGenerateCloudConfigExecutionChartFunc returns a function that renders the gardener-resource-manager configuration for
+// the cloud config user data of the given worker.
+func (b *Botanist) getGenerateCloudConfigExecutionChartFunc(releaseName string, worker gardencorev1beta1.Worker, tokenSecret *corev1.Secret) func() (*chartrenderer.RenderedChart, error) {
```

Comment:
I did this because `generateCloudConfigExecutionChart` requires some input parameters and I want to make the function signature to be consistent with other `generate<Name>Chart` funcs. So I used this function to return a `func() (*chartrenderer.RenderedChart, error)` which would capture the inputs from the context.